File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Power/AngerPower.ts
import { ApplyPowerAction } from "../Action/Common/ApplyPowerAction";
import { CardType } from "../Cards/AbstractCard";
import DungeonManager from "../DungeonManager";
import { OnUseCardMessage } from "../Events/OnUseCardMessage";
import FileHelper from "../FileHelper";
import { LocalizationProperty } from "../Gen/DB/Localization";
import { StringHelper } from "../StringHelper";
import { AbstractPower, PowerType } from "./AbstractPower";
import { StrengthPower } from "./StrengthPower";


export class AngerPower extends AbstractPower{
    Icon: string = FileHelper.FormatPath("Powers/anger.png");
    Id: string = "Anger";
    Name: string = LocalizationProperty.Read("能力-激怒");
    get Type(): PowerType {
        return PowerType.Buff;
    }

    GetDescription(...args: any[]): string {
        return StringHelper.FormatColorString(LocalizationProperty.Read("能力-激怒描述")).format(this.Amount);
    }
    
    OnInit() {
        super.OnInit();
        DungeonManager.MessageManager.Add(OnUseCardMessage.Id,this.OnUseCard)
    }
    
    OnRemove() {
        super.OnRemove();
        DungeonManager.MessageManager.Remove(OnUseCardMessage.Id,this.OnUseCard)
    }
    
    public OnUseCard(msg:OnUseCardMessage){
        if(msg.Card.Type == CardType.SKILL){
            this.AddToTop(new ApplyPowerAction(this.Owner,this.Owner,new StrengthPower(),this.Amount));
            this.Flash();
        }
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Power/Buff/AngerPower.cs
using ModTheSpire.Card;
using ModTheSpire.Dungeon;
using ModTheSpire.Effect.Other;
using ModTheSpire.Event.Ids;
using TCG.Config;

namespace ModTheSpire.Power.Buff
{
    public class AngerPower : AbstractPower
    {
        public override string Id { get; } = nameof(AngerPower);
        public override string Name { get; } = LocalizationProperty.Read("能力-激怒");
        public override string Icon { get; } = "Powers/anger.png";
        public override PowerType Type { get; } = PowerType.Buff;
        public override string GetDescription(object[] args)
        {
            return string.Format(LocalizationProperty.Read("能力-激怒描述"), this.Amount);
        }

        public override void OnInit()
        {
            base.OnInit();
            DungeonManager.MessageManager.Add<OnUseCardMessage>(OnUseCard);
        }

        public override void OnRemove()
        {
            base.OnRemove();
            DungeonManager.MessageManager.Remove<OnUseCardMessage>(OnUseCard);
        }
        
        private void OnUseCard(OnUseCardMessage e)
        {
            if(e.Card.Type == CardType.SKILL){
                this.AddToTop(new ApplyPowerAction(this.Owner,this.Owner,new StrengthPower(),this.Amount));
                this.Flash();
            }
        }
    }
}